File: examples/node-worker/worker.js
import { Worker } from "@cordum/sdk";

const worker = new Worker({
  pool: "hello-pack",
  subjects: ["job.hello-pack.echo"],
  capabilities: ["hello-pack.echo"],
  natsUrl: process.env.NATS_URL || "nats://localhost:4222",
});

worker.handle("job.hello-pack.echo", async (ctx) => {
  const message = ctx.input?.message ?? "hello from node";
  const author = ctx.input?.author ?? "";
  return { message, author };
});

worker.start();

File: examples/python-worker/worker.py
from cordum import Worker, JobContext

worker = Worker(
    pool="hello-pack",
    subjects=["job.hello-pack.echo"],
    capabilities=["hello-pack.echo"],
)


@worker.handler("job.hello-pack.echo")
async def handle_echo(ctx: JobContext):
    message = ctx.input.get("message", "hello from python")
    author = ctx.input.get("author", "")
    return {"message": message, "author": author}


if __name__ == "__main__":
    worker.run()
